lib/rubocop/cop/team.rb

```diff
@@ -174,6 +174,9 @@ def support_target_ruby_version?(cop)
       end
 
       def support_target_rails_version?(cop)
+        # In this case, the rails version was already checked by `#excluded_file?`
+        return true if defined?(RuboCop::Rails::TargetRailsVersion::USES_REQUIRES_GEM_API)
```

Comment:
It looks like the versioning between `rubocop` and `rubocop-rails` gets a bit tricky here.

1. This new version of `rubocop` needs to support older versions of `rubocop-rails`.
2. Inversely, the new version of `rubocop-rails` needs to support older versions of `rubocop`.

I've tested both scenarios, and they both work well.

I added this `USES_REQUIRES_GEM_API` constant to detect new `rubocop-rails` versions, because it was more explicit/clear than checking against some specific version number.

In the future, `TargetRailsVersion` can just be a wrapper for `requires_gem`. When that happens, any cops that don't support the target's rails version would have already be filtered out by `#excluded_file?` called by `#roundup_relevant_cops` above.

For older versions of `rubocop-rails`, that won't be correct, and we'll still need to call their `support_target_rails_version?` method.